### PR TITLE
Allow append of new exceptions to rules

### PR DESCRIPTION
**Pull request by sai-arigeli**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> If contributing rules or changes to rules, please make sure to also uncomment one of the following line:

> /kind rule-update

> /kind rule-create

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area engine

/area rules

> /area tests

> /area proposals

**What this PR does / why we need it**:
PR introduces appending of new fully valid exception objects to existing rules.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update: allow append of new exceptions to rules
```

### CODE REVIEW

**Review by leodido — COMMENTED** [review]
Before reviewing... Why the commit comes from a GitHub user that sign-offs commit with @mstemm email?

**Comment by leodido** [comment]
I refer to this:

image

**Comment by leodido** [comment]
The DCO (implemented via the sign-off) requires the contributor's email (not another one) AFAIK

**Comment by sai-arigeli** [comment]
> The DCO (implemented via the sign-off) requires the contributor's email (not another one) AFAIK

@leodido I had wrongly infered that sign-off needs to be one of the owner. I corrected the commit message.

**Comment by leodido** [comment]
@sai-arigeli I was not understanding because looking at your fork/other PRs via the GitHub events I saw that you were signing-off correctly there (sysdig or gmail email)...

image

File: userspace/engine/lua/rule_loader.lua [hunk 1]
```diff
@@ -592,41 +592,75 @@ function load_rules_doc(rules_mgr, doc, load_state)
...
+			
+			local valid, err
+			if type(eitem['fields']) == "table" then
+			   valid, err = validate_exception_item_multi_fields(rules_mgr, v['source'], eitem, v['context'])
+			else
+			   valid, err = validate_exception_item_single_field(rules_mgr, v['source'], eitem, v['context'])
+			end
+			
+			if valid == false then
+			   return valid, err
```
[the remark this answers is not in the document]
Comment (sai-arigeli) [reply userspace/engine/lua/rule_loader.lua]: @mstemm the 'err' here is already built with context inside the called rule validate functions. I've added 'warnings' to return parameters.

[the remark this answers is not in the document]
Comment (mstemm) [reply userspace/engine/lua/rule_loader.lua]: Great thx!

**Review by mstemm — APPROVED** [review]

**Comment by leogr** [comment]
Closing and reopening to trigger the CI
/close

**Review by mstemm — CHANGES_REQUESTED** [review]
I thought of another change--since this changes the behavior of how falco reads rules files, you should bump the falco engine version here: https://github.com/falcosecurity/falco/blob/master/userspace/engine/falco_engine_version.h#L19. Can you increment it to 11?

**Comment by sai-arigeli** [comment]
> I thought of another change--since this changes the behavior of how falco reads rules files, you should bump the falco engine version here: https://github.com/falcosecurity/falco/blob/master/userspace/engine/falco_engine_version.h#L19. Can you increment it to 11?

@mstemm Fixed this!

**Comment by leogr** [comment]
> > I thought of another change--since this changes the behavior of how falco reads rules files, you should bump the falco engine version here: https://github.com/falcosecurity/falco/blob/master/userspace/engine/falco_engine_version.h#L19. Can you increment it to 11?
> 
> @mstemm Fixed this!

Hey @sai-arigeli 

as per our policy, merge commits are not allowed. Could you use `git rebase` to fix this issue? Thanks in advance

**Comment by sai-arigeli** [comment]
@mstemm @leogr Addressed the change requests.

**Review by mstemm — APPROVED** [review]

**Review by leogr — APPROVED** [review]
LGTM
/approve

**Pull request merged**